experimental/iree/run_benchmarks.py

```diff
@@ -0,0 +1,192 @@
+                   iree_run_module_path: pathlib.Path,
+                   atol: float,
+                   num_threads: str,
+                   verbose: bool = False) -> bool:
+  module_path = artifact_dir / f"module.vmfb"
+  output_npy = artifact_dir / "outputs_npy" / "output_0.npy"
+  command = [
+      str(iree_run_module_path),
+      f"--module={module_path}",
+      f"--task_topology_group_count={num_threads}",
+      "--device=local-task",
+      "--function=main",
+      f"--expected_output=@{output_npy}",
+      f"--expected_f32_threshold={atol}",
+      f"--expected_f16_threshold={atol}",
+      f"--expected_f64_threshold={atol}",
+  ]
+
+  inputs_dir = artifact_dir / "inputs_npy"
+  num_inputs = len(list(inputs_dir.glob("*.npy")))
+  for i in range(num_inputs):
+    command.append(f"--input=@{inputs_dir}/input_{i}.npy")
+
+  command_str = " ".join(command)
+  print(f"Running command: {command_str}")
+
+  try:
+    output = subprocess.run(command, check=True, capture_output=True)
+    if verbose:
+      print(output.stdout.decode())
+    return True
+  except subprocess.CalledProcessError as e:
+    print(f"Error running command: {e}")
+
+  return False
+
+
+def benchmark_one(benchmark: def_types.BenchmarkCase,
+                  target_device: def_types.DeviceSpec,
+                  artifact_dir: pathlib.Path,
+                  iree_benchmark_module_path: pathlib.Path, num_threads: str,
+                  verbose: bool) -> utils.BenchmarkResult:
+  model = benchmark.model
+  benchmark_definition = {
+      "benchmark_name": benchmark.name,
+      "model_name": model.name,
+      "framework": str(model.model_impl.framework_type),
+      "device": target_device.name,
+      "num_threads": num_threads,
+      "tags": model.model_impl.tags + model.tags,
+  }
+
+  inputs_dir = artifact_dir / "inputs_npy"
+  num_inputs = len(list(inputs_dir.glob("*.npy")))
+
+  module_path = artifact_dir / "module.vmfb"
+  command = [
+      str(iree_benchmark_module_path),
+      f"--module={module_path}",
+      f"--task_topology_group_count={num_threads}",
+      "--device=local-task",
+      "--function=main",
+      "--print_statistics",
+  ]
+
+  for i in range(num_inputs):
+    command.append(f"--input=@{inputs_dir}/input_{i}.npy")
+
+  metrics = benchmark_lib.run_benchmark_command(" ".join(command), verbose)
+  return utils.BenchmarkResult(
+      definition=benchmark_definition,
+      metrics={
+          "compiler_level": metrics,
+      },
+  )
+
+
+def _parse_arguments() -> argparse.Namespace:
+  parser = argparse.ArgumentParser(description="Runs IREE benchmarks.")
+  parser.add_argument("-o",
+                      "--output",
+                      type=pathlib.Path,
+                      required=True,
+                      help="JSON file path to merge the results.")
+  parser.add_argument(
+      "--artifact_dir",
+      type=pathlib.Path,
+      required=True,
+      help="The directory containing all required benchmark artifacts.")
```

Comment:
I think ideally we can support running benchmarks without `artifact_dir`, which means that we call function in `compile_workloads` to download and compile models with `iree-compile`. This is convenient for local run. Can be done in the future.

Reply:
That makes sense. A lot of this flow is based on making it work for Android and the benchmark infra most optimally. This means compiling all the models we want to benchmark first, then pushing them to device, then benchmarking. I'll add an issue.